pyiron_base/project/generic.py

```diff
@@ -174,6 +175,16 @@ def name(self):
     def create(self):
         return self._creator
 
+    @property
+    def browser(self):
+        """Provides a browser to inspect the data system of the project."""
+        if self._project_browser is None:
+            from pyiron_base.project.project_browser import ProjectBrowser
```

Comment:
Nit: move import to top of file

Reply:
Ahaaaa, is this down here to avoid circular import? Doesn't it complain about circular imports anyway? If moving it down here is protecting from the circular import somehow I am still concerned this is a yellow flag to rethink the architecture so such a workaround is not needed.

Reply:
It does not complain on circular imports or the like. I just tried to not overload the `Project` import and I just don't see a reason why a `Project` should require `ipywidgets` and `IPython` to run. Of course I could also wrap this in an `@import_alarm` and tell the user to install these to have the browser functionality.

Reply:
> Of course I could also wrap this in an @import_alarm and tell the user to install these to have the browser functionality.

I like this solution best

Reply:
Oh, I just re-tested and it is a circular import... And I just need to import to check if it `isinstance(BaseProject)`.

Reply:
Try if it works with [relative imports](https://docs.python.org/3/reference/import.html#relativeimports) I seem to vaguely recall that this sometimes help with circular imports.

Reply:
I tried this, however, it does not seem to help for the circular import. What actually helps is to `import pyiron_base.project.generic` in the `project_browser.py`. I think each time one uses `from some.module import SomeClass` the `some.module` is init to get `SomeClass`. Just providing the `some.module` as namespace does not need a init at this point which breaks the circular import. 
In the present case, however, I would rather stick to the old implementation since otherwise we would need to have a `@import_warning` on the `Project` level which I do not like at all.

Reply:
Now, I am using `static_isinstance` and do not need to import any `Project` in the `ProjectBrowser`.